plugin/src/helpers/config.ts

```diff
@@ -200,20 +200,47 @@ export async function modifyConfig({
   netlifyConfig,
   cacheDir,
   neededFunctions,
+  isDev,
 }: {
   netlifyConfig: NetlifyConfig
   cacheDir: string
   neededFunctions: FunctionList
+  isDev?: boolean
 }): Promise<void> {
   mutateConfig({ netlifyConfig, cacheDir, neededFunctions })
 
-  if (neededFunctions.includes('API')) {
-    // Editing _redirects so it works with ntl dev
+  const redirectsFileName = join(netlifyConfig.build.publish, '_redirects')
+
+  await spliceConfig({
+    startMarker: '# @netlify/plugin-gatsby redirects start',
+    endMarker: '# @netlify/plugin-gatsby redirects end',
+    contents: neededFunctions.includes('API')
+      ? '/api/* /.netlify/functions/__api 200'
+      : '',
```

Comment:
this now always handles redirects - if API is "not needed" - it will clean up dangling redirect